src/Plugin/RulesAction/SystemMailToUsersOfRole.php

```diff
@@ -0,0 +1,156 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\rules\Plugin\RulesAction\SystemMailToUsersOfRole.
+ */
+
+namespace Drupal\rules\Plugin\RulesAction;
+
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\Core\Mail\MailManagerInterface;
+use Drupal\rules\Core\RulesActionBase;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Drupal\Core\Config\ImmutableConfig;
+use Drupal\user\UserStorage;
+
+/**
+ * Provides a 'Mail to users of a role' action.
+ *
+ * @RulesAction(
+ *   id = "rules_mail_to_users_of_role",
+ *   label = @Translation("Mail to users of a role"),
+ *   category = @Translation("System"),
+ *   context = {
+ *     "roles" = @ContextDefinition("entity:user_role",
+ *       label = @Translation("Roles"),
+ *       description = @Translation("The roles to which to send the e-mail."),
+ *       multiple = TRUE
+ *     ),
+ *     "subject" = @ContextDefinition("string",
+ *       label = @Translation("Subject"),
+ *       description = @Translation("The subject of the e-mail."),
+ *     ),
+ *     "body" = @ContextDefinition("string",
+ *       label = @Translation("Body"),
+ *       description = @Translation("The body of the e-mail."),
+ *     ),
+ *     "from" = @ContextDefinition("email",
+ *       label = @Translation("From"),
+ *       description = @Translation("The from e-mail address."),
+ *       required = FALSE
+ *     )
+ *   }
+ * )
+ *
+ * @todo: Add access callback information from Drupal 7.
+ */
+class SystemMailToUsersOfRole extends RulesActionBase implements ContainerFactoryPluginInterface {
+
+  /**
+   * @var \Psr\Log\LoggerInterface
+   */
```

Comment:
Can we add docs to all this class variables? See https://www.drupal.org/node/1354#var